extensions/amp-video-service/0.1/amp-video-service.js

```diff
@@ -76,14 +148,159 @@ export class VideoService {
 
   /**
    * @param {!AmpElement} unusedVideo
-   * @param {!../../../src/observable.Observable<boolean>} unusedObservable
+   * @param {!Observable<boolean>} unusedObservable
    */
   delegateAutoplay(unusedVideo, unusedObservable) {
     warnUnimplemented('Autoplay delegation');
   }
 }
 
 
+/** @visibleForTesting */
+export class VideoEntry {
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!VideoService} videoService
+   * @param {!../../../src/video-interface.VideoInterface} video
+   */
+  constructor(ampdoc, videoService, video) {
+
+    /** @private @const{!../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
+
+    /** @private @const {!VideoService} */
+    this.service_ = videoService;
+
+    /** @private @const{!../../../src/video-interface.VideoInterface} */
+    this.video_ = video;
+
+    /** @visibleForTesting {boolean} */
+    this.isPlaying = false;
+  }
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!VideoService} videoService
+   * @param {!../../../src/video-interface.VideoInterface} video
+   */
+  static create(ampdoc, videoService, video) {
```

Comment:
I like this, we don't use this pattern often enough.

Reply:
:)